src/System.Management.Automation/engine/parser/CharTraits.cs

```diff
@@ -374,13 +374,26 @@ internal static bool ForceStartNewToken(this char c)
             return c.IsWhitespace();
         }
 
-        // Return true if the character ends the current number token.  This allows the tokenizer
-        // to scan '7z' as a single token, but '7+' as 2 tokens.
-        internal static bool ForceStartNewTokenAfterNumber(this char c)
+        /// <summary>
+        /// Check if the current character forces to end scanning a number token.
+        /// This allows the tokenizer to scan '7z' as a single token, but '7+' as 2 tokens.
+        /// </summary>
+        /// <param name="c">The character to check.</param>
+        /// <param name="forceEndNumberOnTernaryOperatorChars">
+        /// In some cases, we want '?' and ':' to end a number token too, so they can be
+        /// treated as the ternary operator tokens.
+        /// </param>
+        /// <returns>Return true if the character ends the current number token.</returns>
+        internal static bool ForceStartNewTokenAfterNumber(this char c, bool forceEndNumberOnTernaryOperatorChars)
         {
             if (c < 128)
             {
-                return (s_traits[c] & CharTraits.ForceStartNewTokenAfterNumber) != 0;
+                if ((s_traits[c] & CharTraits.ForceStartNewTokenAfterNumber) != 0)
+                {
+                    return true;
+                }
+
+                return forceEndNumberOnTernaryOperatorChars && (c == '?' || c == ':');
```

Comment:
Why cannot `?` and `:` just be added to `CharTraits.ForceStartNewTokenAfterNumber`?  This trait is only used while the tokenizer is in expression mode.

Reply:
I might answer my own question …  because it needs to be hidden behind an experimental feature flag at this time?

Reply:
I don't think the PR context explained clearly enough what this is for:

(if the following are entire command lines)
`1?2:3` is a command
`1-lt2?3:4` is a ternary expression.

It will be even harder for EditorSyntax (https://github.com/PowerShell/EditorSyntax/issues/184) to correctly syntax scope this via TextMate.  This will require an extra set of `numericConstant` rules to handle the 'first possible operand of an expression' vs 'a numeric value clearly in expression mode' vs 'a numeric constant in command mode'.  However, it is an interesting challenge to see if I can simplify.

Reply:
Comments were added for the parameter `forceEndNumberOnTernaryOperatorChars`.
```
        /// <param name="forceEndNumberOnTernaryOperatorChars">
        /// In some cases, we want '?' and ':' to end a number token too, so they can be
        /// treated as the ternary operator tokens.
```
Also, comments were added in `Parser.cs` to the places where the parameter `endNumberOnTernaryOpChars` was added.